Lower bound on `conduit` is too lax

…which seems to be the reason for the Hackage docbuilder failure at https://hackage.haskell.org/package/persistent-2.14.0.3/reports/2:

```
Database/Persist/SqlBackend/Internal/Statement.hs:8:1: error:
    Could not find module Conduit
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
  |
8 | import Conduit
  | ^^^^^^^^^^^^^^
```

The `Conduit` module was introduced in [`conduit-1.3.0`](https://hackage.haskell.org/package/conduit-1.3.0/changelog), but `persistent`'s bound is `conduit >=1.2.12`.

--- a/persistent/persistent.cabal
+++ b/persistent/persistent.cabal
@@ -22,7 +22,7 @@ library
       , base64-bytestring
       , blaze-html               >= 0.9
       , bytestring               >= 0.10
-      , conduit                  >= 1.2.12
+      , conduit                  >= 1.3
       , containers               >= 0.5
       , deepseq
       , fast-logger              >= 2.4

